openrouteservice/src/main/java/org/heigit/ors/routing/graphhopper/extensions/weighting/MaximumSpeedWeighting.java

```diff
@@ -0,0 +1,120 @@
+/*  This file is part of Openrouteservice.
+ *
+ *  Openrouteservice is free software; you can redistribute it and/or modify it under the terms of the
+ *  GNU Lesser General Public License as published by the Free Software Foundation; either version 2.1
+ *  of the License, or (at your option) any later version.
+
+ *  This library is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ *  without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *  See the GNU Lesser General Public License for more details.
+
+ *  You should have received a copy of the GNU Lesser General Public License along with this library;
+ *  if not, see <https://www.gnu.org/licenses/>.
+ */
+package org.heigit.ors.routing.graphhopper.extensions.weighting;
+
+import com.graphhopper.routing.profiles.DecimalEncodedValue;
+import com.graphhopper.routing.util.FlagEncoder;
+import com.graphhopper.routing.util.HintsMap;
+import com.graphhopper.routing.weighting.Weighting;
+import com.graphhopper.util.EdgeIteratorState;
+import com.graphhopper.util.Parameters.Routing;
+import org.heigit.ors.config.AppConfig;
+
+/**
+ * This class creates the weighting for the routing according to the maximum speed set by user.
+ *
+ * @author Athanasios Kogios
+ */
+
+public class MaximumSpeedWeighting implements Weighting {
+    protected final static double SPEED_UNIT_CONVERTER = 3.6; //From km/h to m/s.
+    private final double headingPenalty;
+    private final  double userMaxSpeed;
+    private final Weighting superWeighting;
+    private final DecimalEncodedValue avSpeedEnc;
+    private boolean calculateWeight;
+    private double maxSpeed =  ((AppConfig.getGlobal().getServiceParameter("routing.profiles.default_params","maximum_speed_lower_bound")) != null)
+            ?   Double.parseDouble(AppConfig.getGlobal().getServiceParameter("routing.profiles.default_params","maximum_speed_lower_bound"))
+            : 80; //If there is a maximum_speed value in the app.config we use that. If not we set a default of 80.
+
+    public MaximumSpeedWeighting(FlagEncoder flagEncoder, HintsMap hintsMap, Weighting weighting) {
+        this.avSpeedEnc = flagEncoder.getAverageSpeedEnc();
+        this.superWeighting = weighting;
+        if(hintsMap.getDouble("user_speed",maxSpeed) < maxSpeed  ){
+            throw new RuntimeException("User speed should be <=" + maxSpeed);
+        }
+        this.headingPenalty = hintsMap.getDouble(Routing.HEADING_PENALTY, Routing.DEFAULT_HEADING_PENALTY);
+        this.userMaxSpeed = hintsMap.getDouble("user_speed",maxSpeed);
+        this.calculateWeight = (hintsMap.getWeighting() == "fastest");
```

Comment:
I think it would be better to set the flag based on `superWeighting.getName()` rather than `hintsMap.getWeighting()`.